Core/include/Acts/Vertexing/SeedVertexFinder.hpp

```diff
@@ -0,0 +1,193 @@
+    // spacepoints
+    float maxXYZdeviation = 0.08;
+
+    // minimum angle between Z axis and a triplet, effectively removing triplets
+    // with large |eta|
+    float minTheta = 1.;
+
+    // thresholds for near, middle, and far spacepoints
+    float rMinNear = 20.f * Acts::UnitConstants::mm;
+    float rMaxNear = 60.f * Acts::UnitConstants::mm;
+    float rMinMiddle = 150.f * Acts::UnitConstants::mm;
+    float rMaxMiddle = 190.f * Acts::UnitConstants::mm;
+    float rMinFar = 280.f * Acts::UnitConstants::mm;
+    float rMaxFar = 320.f * Acts::UnitConstants::mm;
+
+    // number of phi slices, at least 3 to avoid duplicities
+    // it should be less than 2*pi/maxPhideviation in order not to loop over
+    // triplets that will be rejected by maxPhideviation anyway
+    int numPhiSlices = 60;
+    // use only a fraction of available phi slices to speed up calculations;
+    float useFracPhiSlices = 0.5;
+
+    // number of z slices
+    int numZSlices = 150;
+    // use only a fraction of available z slices to speed up calculations;
+    float useFracZSlices = 0.5;
+    // maximum |z| to consider, z slices will be done within the range
+    // (-maxAbsZ,maxAbsZ) values of maxAbsZ, maxZPosition, rMaxFar, and minTheta
+    // should be set reasonably with respect to each other
+    float maxAbsZ = 450. * Acts::UnitConstants::mm;
+
+    // maximum Z position of the vertex at the point closest to the Z axis
+    float maxZPosition = 200.f * Acts::UnitConstants::mm;
+    // maximum R position of the vertex at the point closest to the Z axis
+    float maxRPosition = 10.f * Acts::UnitConstants::mm;
+
+    // chi^2 minimalization will happen with respect to "planes" or "rays"
+    std::string minimalizeWRT = "planes";
+
+    // maximum number of iterations when discarding triplets with the largest
+    // chi^2
+    int maxIterations = 20;
+    // each iteration, discard this fraction of triplets with the largest chi^2
+    float removeFraction = 0.10;
+    // if the vertex estimation moves less than this, stop iterations
+    float minVtxShift = 0.3f * Acts::UnitConstants::mm;
+  };
+
+  /// Const access to the config
+  const Config& config() const { return m_cfg; }
+
+  /// @brief Constructor
+  ///
+  /// @param cfg Configuration object
+  /// @param lgr Logging instance
+  SeedVertexFinder(const Config& cfg,
+                   std::unique_ptr<const Logger> lgr =
+                       getDefaultLogger("SeedVertexFinder", Logging::DEBUG));
+
+  /// @brief Destructor
+  ~SeedVertexFinder() = default;
+
+  /// @brief Finds the vertex based on the provided spacepoints
+  /// @param spacepoints Vector of the input spacepoints; they do not need to be sorted anyhow
+  /// @return Position of the vertex along z-axis
+  Acts::Vector3 findVertex(const std::vector<spacepoint_t>& spacepoints);
+
+ private:
+  /// @brief Struct to store spacepoint combinations from near, middle, and far parts of the detector
+  struct Triplet {
```

Comment:
Is there any way we can harmonise this with, e.g. the Triplet definition from seeding?

Reply:
Hi @asalzburger, unfortunatelly neither of the triplet definitions works well here.
Moreover, we might be able to get rid of it completely but I would leave it for the next iteration.